doc/conditional_independence.rst

```diff
@@ -0,0 +1,180 @@
+of the test statistic. In both approaches a pvalue is computed, which is compared to :math:`\alpha`. The pvalue
+states the "probability of observing a test-statistic at least as extreme as our observed test-statistic in null distribution". By rejecting
+the null hypothesis, one claims that :math:`X \not\perp Y | Z`, so that X and Y are in fact (conditionally)
+dependent given Z. Note that if one fails to reject the null hypothesis, it is simply by convention that we
+claim X and Y are conditionally independent. It is not necessarily the case, and it is plausible that there
+is a weak dependency that is unable to be captured by our proposed CI test, and/or data samples.
+
+It is because of this reason, one would typically like the most powerful test given assumptions about
+the data. With that in mind, there are various approaches to CI testing that are typically more powerful
+with certain assumptions on the underlying data distribution.
+
+Conditional Mutual Information
+------------------------------
+Conditional mutual information (CMI) is a general formulation of CI, where CMI is defined as
+:math::
+    
+    \\int log \frac{p(x, y | z)}{p(x | z) p(y | z)}
+
+As we can see, CMI is equal to 0, if and only if :math:`p(x, y | z) = p(x | z) p(y | z)`, which
+is exactly the definition of CI. CMI is completely non-parametric and thus requires no assumptions
+on the underlying distributions. Unfortunately, CMI is notoriously difficult to estimate. There are
+various proposals in the literature for estimating CMI, which we summarize here:
+
+- The Kraskov, Stogbauer and Grassberger (KSG) estimate approach estimates mutual information
+  via k-NN statistics :footcite:`kraskov_estimating_2004`. It was generalized to CMI by :footcite:`frenzel_partial_2007`.
+  This class of estimators is asymptotically correct, meaning if we had an infinite amount of data
+  we would obtain the true value of the CMI. However, it relies on statistics generated from the k-NN,
+  which if we implement the naive approach using a KDTree, then it generally suffers in high-dimensions.
+  In our examples, we see it suffer with dimensionality > 4 or 5.
+
+  Estimates of CMI can be converted into a CI hypothesis test by permutation testing :footcite:`Runge2018cmi`.
+  One can generate estimated samples from the null distribution by permuting samples in an intelligent manner
+  and then the CMI value generated from the actual observed data can be compared to the CMI values computed
+  on the permutated datasets to estimate a pvalue.
+
+  It is worth noting that if one has "robust" estimates of k-NN using for example a model that
+  is effective in high-dimensions, then the KSG estimator for CMI may be quite good. For example,
+  one can use variants of Random Forests to generate adaptive nearest-neighbor estimates in high-dimensions
+  or on manifolds, such that the KSG estimator is still quite good.
```

Comment:
I would remove the "quite good"s here. Just say that it can be 'powerful' even in high dimensions if the knn estimator can capture it.

Reply:
Done.